src/core/attribution/nodeScore.js

```diff
@@ -26,3 +27,34 @@ export function scoreByMaximumProbability(
   }
   return scoreMap;
 }
+
+/* Normalize scores so that a group of nodes have a fixed total score */
+export function scoreByConstantTotal(
+  pi: NodeDistribution,
+  totalScore: number,
+  nodeFilter: NodeAddressT /* Normalizes based on nodes matching this prefix */
+): NodeScore {
+  if (totalScore <= 0) {
+    throw new Error("Invalid argument: totalScore must be >= 0");
+  }
+
+  let unnormalizedTotal = 0;
+  for (const [addr, prob] of pi) {
+    if (NodeAddress.hasPrefix(addr, nodeFilter)) {
+      unnormalizedTotal += prob;
+    }
+  }
+  if (unnormalizedTotal === 0) {
+    throw new Error(
+      "Tried to normalize based on nodes with no score. " +
+        "This probably means that there were no nodes matching prefix: " +
```

Comment:
FYI, I just hit this on a real repo (wchargin/varese), because it has no
GitHub activity. No change requested, but we should note that this can
happen even if you don’t hack up your graphs manually.